tests/components/harmony/test_init.py

```diff
@@ -0,0 +1,72 @@
+"""Test init of Logitch Harmony Hub integration."""
+from homeassistant.components.harmony.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers import entity_registry
+from homeassistant.setup import async_setup_component
+
+from .const import (
+    ENTITY_NILE_TV,
+    ENTITY_PLAY_MUSIC,
+    ENTITY_WATCH_TV,
+    HUB_NAME,
+    NILE_TV_ACTIVITY_ID,
+    PLAY_MUSIC_ACTIVITY_ID,
+    WATCH_TV_ACTIVITY_ID,
+)
+
+from tests.common import MockConfigEntry, mock_registry
+
+
+async def test_unique_id_migration(mock_hc, hass, mock_write_config):
+    """Test migration of switch unique ids to stable ones."""
+    entry = MockConfigEntry(
+        domain=DOMAIN, data={CONF_HOST: "192.0.2.0", CONF_NAME: HUB_NAME}
+    )
+
+    entry.add_to_hass(hass)
+    mock_registry(
+        hass,
+        {
```

Comment:
I'm not sure that the `RegistryEntry` will be stable to use in tests, but it looks like it was used in #32417 so it should be ok

It might be safe to use `async_get_or_create` on the `EntityRegistry` instead.